percona: update packages, init 8.4

osnyx opened the pull request:
- updates LTS packages for percona-server and percona-xtrabackup in the 8.0 release series
- inits the 8.4 release series of percona-server and percona-xtrabackup
  - the 8.4 series is another LTS release, hence the `percona-server_lts` and `percona-xtrabackup_lts` now point towards these packages
  - the 8.3 innovation series is now end-of-life, thus the `-innovation` package names temporarily point to 8.4 as well, until the release of 8.5

I had to restructure the package naming again, partially undoing #309953:
When reworking the percona package naming towards the two release
streams "innovation" and "lts", I did not anticipate the case that two
releases are long-term supported in parallel, and that the latest
Innovation release is end-of-life and replaced by a more recent lts.

Hence, the first-level attribute names now reflect the package
major-minor versions again, augmented by attribute names pointing to
respective lts and innovation releases.

### notes on backporting

I intend to manually backport this to release-24.05. The package-updates will be kept as-is, but the `-lts` names will continue to point to the 8.0 release series to avoid breaking changes within a release.
The backport is necessary, as 8.3 is end-of-life by now.

**Changelogs**:
- https://docs.percona.com/percona-xtrabackup/8.0/release-notes/8.0/8.0.35-31.0.html#release-highlights
- https://docs.percona.com/percona-xtrabackup/8.4/release-notes/8.4.0-1.html
- https://docs.percona.com/percona-server/8.0/release-notes/8.0.37-29.html
- https://docs.percona.com/percona-server/8.4/release-notes/8.4.0-1.html

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

osnyx committed:
percona-server_8_4: init at 8.4.0-1

This is a new LTS release as well.

osnyx committed:
percona-xtrabackup_8_0: 8.0.36-30 -> 8.0.35-31

osnyx committed:
percona-xtrabackup_8_4: init at 8.4.0-1

osnyx committed:
percona-server, percona-xtrabackup: rework naming

When reworking the percona package naming towards the two release
streams "innovation" and "lts", I did not anticipate the case that two
releases are long-term supported in parallel, and that the latest
Innovation release is end-of-life and replaced by a more recent lts.

Hence, the first-level attribute names now reflect the package
major-minor versions again, augmented by attribute names pointing to
respective lts and innovation releases.

osnyx committed:
percona: release notes for 8.4 LTS release

osnyx committed:
percona: apply nixfmt

osnyx committed:
.git-blame-ignore-revs: add percona reformatting

osnyx commented:
I can also offer to run this through our extended test suite that combines percona-server and xtrabackup invocations, but that might take a few days.

osnyx commented:
I found a typo in a commit message and had to reword that commit. Code remains unchanged.

osnyx commented:
Our internal extended test suite currently fails for `percona-server_8_4`, but this is likely cause by native password authentication now being disabled by default.
I added this change to the NixOS 24.11 release notes.

For `percona-server_8_0` the test suite still passes.

osnyx commented:
I can happily report that our [extended test suite](https://github.com/flyingcircusio/fc-nixos/blob/fc-24.11-dev/tests/mysql.nix) passes again after enabling legacy password auth again. So the packages are working fine.

Ma27 reviewed (changes requested):
The changes are looking good and the packages build for me.

Two things however:

* Should we also update the list of percona packages in `nixos/tests/mysql/common.nix` to match our current naming scheme? I'd suggest to just do `percona-server_8_0` and `percona-server_8_4` there (and `percona-server_8_5` when it gets out - or 9_0 or whatever the next one will be).
* Also, I have one request about the commits themselves: I see that you've renamed files in the commits and then changed the imports in another commit (the rework naming part). This has the downside that there are some commits where percona doesn't evaluate which has the potential of making bisects unnecessarily hard. Can you please make sure that if you rename a file in a commit, the import of the file also gets changed already?

osnyx committed:
percona-server_8_0: 8.0.36-28 -> 8.0.37-29

This is a new LTS release as well.

leona-ya commented on pkgs/top-level/all-packages.nix:
```diff
@@ -11089,9 +11089,15 @@ with pkgs;
+  inherit (import ../tools/backup/percona-xtrabackup pkgs) percona-xtrabackup_8_0 percona-xtrabackup_8_4;
+  percona-xtrabackup_lts = percona-xtrabackup_8_4;
+  # temporarily, latest LTS and Innovation release are equal
+  percona-xtrabackup_innovation = percona-xtrabackup_8_4;
```
I'd vote for this to be in `aliases.nix`.

osnyx replied:
Thanks for catching this, I did not see any announcements in just the release notes.

leona-ya reviewed (commented):
Generally this seems to be the right move. Also because Percona [won't release Innovation releases anymore for 9.x](https://www.percona.com/blog/no-mysql-9-x-innovation-releases-from-percona/)

osnyx committed:
percona: deprecate innovation/lts release flavours

Percona upstream has decided not to create releases or each MySQL
innovation upstream release, but just for LTS releases.
https://www.percona.com/blog/no-mysql-9-x-innovation-releases-from-percona/

osnyx commented:
@Ma27 You're correct with your remarks on rebasing. My motivation was to lay out the commits in a way that makes backporting easier, but I guess I found another good commit scoping now.

osnyx committed:
tests/mysql: properly specify percona packages

osnyx committed:
percona: correct licensing information

Corrects the partially wrong change from #313931, all Oracle MySQL
derivates are actually GPLv2 only.

osnyx commented:
I fixed some small further nitpicks, also updated the designated backport. Please re-review.

jopejoe1 commented:
The forgot to change the passtru.test to the new package name, made a fix for that in #349757
